### PR TITLE
esp32/i2s: implement I2S receiver module

### PR DESCRIPTION
**Pull request by tmedicci**
## Summary

* Add ioctl method to enable allocating the apb buffer.
* Add RX methods to set data width, sample rate, channels and or receiving data from the I2S peripheral.
* Update the i2schar defconfig to enable the I2S receiver.
* Add nxlooper defconfig to enable testing the RX interface.
* Add specific bindings on ESP32 bringup to enable nxlooper to work without the need of any specific codec.

## Impact

No impact regarding general usage.

## Testing

Internal CI + ESP32-DevKitC board as master (receiving I2S data frame and reproducing it) and ESP32S2-Saola-1 board providing the original I2S data frame (set-up as TX slave and attached to the ESP32-DevKitC I2S RX).

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/xtensa/esp32/common/src/esp32_board_i2sdev.c [hunk 1]
```diff
@@ -54,58 +54,101 @@
...
+    }
+
+  if (enable_rx)
     {
-      auderr("ERROR: Failed to register /dev/%s device: %d\n", devname, ret);
+      /* Initialize audio input */
+
+      audio_i2s = audio_i2s_initialize(i2s, false);
+
+      if (!audio_i2s)
```
Comment (pkarashchenko) [inline boards/xtensa/esp32/common/src/esp32_board_i2sdev.c:136]:
```suggestion
      if (audio_i2s == NULL)
```

File: boards/xtensa/esp32/common/src/esp32_board_i2sdev.c [hunk 2]
```diff
@@ -54,58 +54,101 @@
...
 
-  if (ret < 0)
+#ifdef CONFIG_SYSTEM_NXLOOPER
+      ret = audio_register(devname, audio_i2s);
+#else
+      struct audio_lowerhalf_s *pcm;
+
+      pcm = pcm_decode_initialize(audio_i2s);
+
+      if (!pcm)
```
Comment (pkarashchenko) [inline boards/xtensa/esp32/common/src/esp32_board_i2sdev.c:113]:
```suggestion
      if (pcm == NULL)
```

File: boards/xtensa/esp32/common/src/esp32_board_i2sdev.c [hunk 3]
```diff
@@ -54,58 +54,101 @@
...
 
-  pcm = pcm_decode_initialize(audio_i2s);
+      audio_i2s = audio_i2s_initialize(i2s, true);
 
-  if (!pcm)
-    {
-      auderr("ERROR: Failed create the PCM decoder\n");
-      return  -ENODEV;
-    }
+      if (!audio_i2s)
```
Comment (pkarashchenko) [inline boards/xtensa/esp32/common/src/esp32_board_i2sdev.c:93]:
```suggestion
      if (audio_i2s == NULL)
```

File: arch/xtensa/src/esp32/esp32_i2s.c [hunk 4]
```diff
@@ -808,6 +918,77 @@ static IRAM_ATTR int i2s_txdma_setup(struct esp32_i2s_s *priv,
...
+  uint32_t bytes_queued;
+  irqstate_t flags;
+
+  DEBUGASSERT(bfcontainer && bfcontainer->apb);
+
+  inlink = bfcontainer->dma_link;
+
+  /* Allocate the internal buffer for RX */
+
+  bfcontainer->buf = (uint8_t *)calloc(bfcontainer->nbytes, 1);
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32/esp32_i2s.c:954]:
```suggestion
  bfcontainer->buf = calloc(bfcontainer->nbytes, 1);
```

**Review by pkarashchenko — APPROVED** [review]
All comment are minor

**Comment by tmedicci** [comment]
> All comment are minor

Thanks for the suggestions, @pkarashchenko !

**Pull request merged**